feature_engine/datetime/datetime.py

```diff
@@ -204,12 +208,33 @@ def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
         # check input dataframe
         X = _is_dataframe(X)
 
-        # find or check for datetime variables
-        self.variables_ = _find_or_check_datetime_variables(X, self.variables)
+        # special case index
+        if self.variables == "index":
 
-        # check if datetime variables contains na
-        if self.missing_values == "raise":
-            _check_contains_na(X, self.variables_)
+            if not (
+                is_datetime(X.index)
+                or (
+                    not is_numeric(X.index) and _is_categorical_and_is_datetime(X.index)
+                )
+            ):
+                raise TypeError("Index is not datetime.")
+
+            if self.missing_values == "raise":
+                if X.index.isnull().any():
+                    raise ValueError(
+                        "Index contain NaN. Check and "
+                        "remove those before using this transformer."
```

Comment:
Could we re-phrase this error message?

"The dataframe index contains missing data. Check and remove those before using this transformer or set `missing_values` to False."